strongbox-data-service/src/main/java/org/carlspring/strongbox/config/DataServiceConfig.java

```diff
@@ -57,48 +47,29 @@
      * This must be after {@link OEntityUnproxyAspect} order.
      */
     public static final int TRANSACTIONAL_INTERCEPTOR_ORDER = 100;
-    
-    @Inject
-    private ConnectionConfig connectionConfig;
-    
-    @Inject
-    private ResourceLoader resourceLoader;
-    
+
     @Inject
-    private OrientDbServer server;
+    private DataSource dataSource;
 
-    @PostConstruct
-    public void init() throws ClassNotFoundException, LiquibaseException
+    @Bean
+    public SpringLiquibase springLiquibase(ResourceLoader resourceLoader)
     {
-        server.start();
-        
         SpringLiquibase liquibase = new SpringLiquibase();
-        liquibase.setDataSource(dataSource());
+        liquibase.setDataSource(dataSource);
         liquibase.setResourceLoader(resourceLoader);
         liquibase.setChangeLog("classpath:/db/changelog/db.changelog-master.xml");
-        liquibase.afterPropertiesSet();
-
-        new TransactionTemplate(transactionManager()).execute((s) -> {
-            OEntityManager oEntityManager = oEntityManager();
-            // register all domain entities
-            Stream.concat(new Reflections("org.carlspring.strongbox").getTypesAnnotatedWith(Entity.class).stream(),
-                          new Reflections("org.carlspring.strongbox").getTypesAnnotatedWith(Embeddable.class).stream())
-                  .forEach(oEntityManager::registerEntityClass);
-            
-            return null;
-        });
-
-
+        return liquibase;
     }
 
     @Bean
-    public PlatformTransactionManager transactionManager()
+    public PlatformTransactionManager transactionManager(EntityManagerFactory emf)
     {
-        return new JpaTransactionManager(entityManagerFactory());
```

Comment:
@fuss86 can't we leave this as it was without `@Bean` method arguments (everywhere in this configuration)? Once I found that Spring better resolves circular/transitive bean dependencies with such approach.

Reply:
If you deliver `@Bean` method argument like in this example: `PlatformTransactionManager transactionManager(EntityManagerFactory emf)` you're expecting `EntityManagerFactory` initialized instance to actually initialize `PlatformTransactionManager` so it's the analogous way and clearer a bit in my opinion.

Reply:
yeah, seems that you reworked the configuration structure, so maybe you are right. I mean that we should resolve bean dependencies with expectedly configuration method calls if possible rather then method arguments injection.